cmd/xl-storage-format-utils.go

```diff
@@ -18,21 +18,24 @@
 package cmd
 
 import (
+	"sort"
+
 	jsoniter "github.com/json-iterator/go"
 )
 
+// versionsSorter sorts FileInfo slices by version.
 type versionsSorter []FileInfo
 
-func (v versionsSorter) Len() int      { return len(v) }
-func (v versionsSorter) Swap(i, j int) { v[i], v[j] = v[j], v[i] }
-func (v versionsSorter) Less(i, j int) bool {
-	if v[i].IsLatest {
-		return true
-	}
-	if v[j].IsLatest {
-		return false
-	}
-	return v[i].ModTime.After(v[j].ModTime)
+func (v versionsSorter) sort() {
```

Comment:
Just tested this to check that we don't need a pointer here :)

Reply:
Correct. Since v is a slice and sorting doesn't change the size, only inside the data pointed to by v, this is safe.